helpers/clientopts.go

```diff
@@ -0,0 +1,12 @@
+package helpers
+
+type RegistrationOpts struct {
+	Localpart string // default '' (don't care)
+	DeviceID  string // default '' (generate new)
+	Password  string // default 'complement_meets_min_password_requirement'
+	IsAdmin   bool   // default false
```

Comment:
Aside: do any of the tests actually use the adminness? It's not a specced property of Matrix as much as a backdoor for Synapse, I thought?

Reply:
Yes tests use it. It _is_ a specced property in Matrix all over the place, just no means to make one via the login/register APIs. See https://spec.matrix.org/latest/client-server-api/#get_matrixclientv3adminwhoisuserid for the most obvious example.